caffe2/operators/hip/conv_op_miopen.hip

```diff
@@ -252,48 +253,101 @@ bool MIOPENConvOp::DoRunWithType() {
     VLOG(1) << "Changing MIOpen descriptor configurations.";
     if (input_changed) {
       mio_input_dims_ = X.sizes().vec();
-      MIOPEN_ENFORCE(miopenSet4dTensorDescriptor(
-          bottom_desc_, miopenTypeWrapper<T_X>::type, N, C, H, W));
+      if (mio_input_dims_.size() == 4) {
+        MIOPEN_ENFORCE(miopenSet4dTensorDescriptor(
+            bottom_desc_, miopenTypeWrapper<T_X>::type, N, C, H, W));
+      } else {
+        vector<int> input_dims (begin(mio_input_dims_), end(mio_input_dims_));
+        MIOPEN_ENFORCE(miopenSetTensorDescriptor(
+            bottom_desc_, miopenTypeWrapper<T_X>::type, input_dims.size(), input_dims.data(), nullptr));
```

Comment:
is nullptr allowed here? MIOpen doc is silent on that.

Reply:
Yes, confirmed with @ce1adon before implementation.